contentcuration/contentcuration/viewsets/contentnode.py

```diff
@@ -676,12 +676,20 @@ def move(self, pk, target=None, position=None):
 
         try:
             target, position = self.validate_targeting_args(target, position)
-            try:
-                contentnode.move_to(target, position)
-            except ValueError:
-                raise ValidationError(
-                    "Invalid position argument specified: {}".format(position)
-                )
+
+            channel_id = target.channel_id
+
+            task_args = {
+                "user_id": self.request.user.id,
+                "channel_id": channel_id,
+                "node_id": contentnode.id,
+                "target_id": target.id,
+                "position": position,
+            }
+
+            task, task_info = create_async_task(
+                "move-nodes", self.request.user, **task_args
+            )
```

Comment:
@rtibbles Does the calling code expect the operation to be completed by the time this function returns? If so, could doing this lead changes to be applied out of order?

Reply:
Move operations are commutative with other changes, except delete - in which case the task will just error.